python/pyspark/sql/tests/test_arrow.py

```diff
@@ -133,18 +133,29 @@ def test_toPandas_fallback_enabled(self):
                         user_warns = [
                             warn.message for warn in warns if isinstance(warn.message, UserWarning)]
                         self.assertTrue(len(user_warns) > 0)
-                        self.assertTrue(
-                            "Attempting non-optimization" in str(user_warns[-1]))
-                        assert_frame_equal(pdf, pd.DataFrame({"a": [[ts]]}))
+                        self.assertTrue("Attempting non-optimization" in str(user_warns[-1]))
+                        assert_frame_equal(pdf, pd.DataFrame({"a": [[[ts]]]}))
 
     def test_toPandas_fallback_disabled(self):
-        schema = StructType([StructField("a", ArrayType(TimestampType()), True)])
-        df = self.spark.createDataFrame([(None,)], schema=schema)
+        ts = datetime.datetime(2015, 11, 1, 0, 30)
+        schema = StructType([StructField("a", ArrayType(ArrayType(TimestampType())), True)])
+        df = self.spark.createDataFrame([([[ts]],)], schema=schema)
         with QuietTest(self.sc):
             with self.warnings_lock:
                 with self.assertRaisesRegex(Exception, 'Unsupported type'):
                     df.toPandas()
 
+    def test_toPandas_array_timestamp(self):
+        schema = StructType([
+            StructField("idx", LongType(), True),
+            StructField("timestamp_array", ArrayType(TimestampType()), True)])
+        data = [(0, [datetime.datetime(1969, 1, 1, 1, 1, 1)]),
+                (2, [datetime.datetime(2100, 3, 3, 3, 3, 3)])]
```

Comment:
The data here is a little too simple, there should be multiple values including nulls

Reply:
Have tested with none . but agree , will add more stringent test cases.